docs/naps/6-contributable-menus.md

```diff
@@ -0,0 +1,517 @@
+The five proposed submenus are `Projection`, `Segmentation`, `Classification`, `Registration` and `Tracks`.
+
+Many of the actions in this menu exist in the right click layer context menu. These items
+should be replicated in the `Layers` menu as needed, both to aid discoverability and
+to ensure users are not met with empty menus on initially opening `napari`. It's
+also possible that exceedingly common operations e.g. thresholds, are provided in the future
+by `napari` itself.
+
+### The `Acquisition` Menu
+
+In addition to the `Layers` menu, we add `Acquisition` as a top level menu.
+
+`Acquisition` will contain widgets and utilities for interfacing with microscopes and
+other types of cameras.
+
+### The I/O Utilities Menu
+
+A cursory analysis of widget names revealed a minimum of 17 plugins provide
+widgets dedicated to importing and exporting of data, features, models and/or
+other material supporting analysis.
+
+These widgets usually require more choices from the user than are currently
+possible via the `reader` and `writer` interfaces. Although many discussions have
+been raised about expanding the opening and saving options in `napari` to 
+support more complex choices ([#1637](https://github.com/napari/napari/issues/1637), 
+[#2801](https://github.com/napari/napari/issues/2801), 
+[#4611](https://github.com/napari/napari/issues/4611), 
+[#4882](https://github.com/napari/napari/pull/4882)...),
+we are not presently close to providing a unified interface for complex file opening/saving.
+Additionally, default `napari` opening and saving is entirely focused on reading data into 
+layer, but there are many other reasons a user may wish to read a file or save some output
+from the `Viewer`. 
+
+It is likely, therefore, that some plugins will always have bespoke interfaces for importing and
+exporting various file formats. These interfaces will be exposed via the new `File->I/O Utilities`
+menu.
+
+### Plugin Submenus
+
+The goal of the newly proposed menus is to provide a natural place where generally applicable
+actions can be semantically organized and easy to locate. However, many `napari` plugins
+contain an assortment of highly specialized widgets (that often interact with each other)
+that support highly specific, and sometimes ordered, workflows and analyses.
+
+It may never make sense for such plugins to distribute their widgets across the different
+`napari` menus, particularly when they are designed to work in concert on specific
+data formats or layer types.
+
+We therefore give plugin developers full control over their own submenu under `Plugins->My Plugin`.
+Plugin developers can organize all contributions under this submenu as they see fit, including adding their own submenus of arbitrary depth.
+
+### Complete Set of Proposed Contributable `napari` Menus
+
+```
+File
+├─ ...
+├─ IO Utilities
+Layers
+├─ Visualization
+├─ Edit
+│  ├─ Annotation Tools
```

Comment:
```suggestion
│  ├─ Annotate
```
I think this is more in line with the other `Edit` items.

Also, I think we should bring `Annotate` out to a top level `Layers` menu (or even it's own category). I think annotation is a pretty commun napari use case and thus should be easy to access. Happy to merge now and discuss later though.

Reply:
Should `Annotate` maybe be under the Edit menu vs. the Layer menu?